setup.py

```diff
@@ -1,88 +1,13 @@
-import os
 import sys
 
-from setuptools import find_packages, setup
-
-
-def read(rel_path: str) -> str:
-    here = os.path.abspath(os.path.dirname(__file__))
-    # intentionally *not* adding an encoding option to open, See:
-    #   https://github.com/pypa/virtualenv/issues/201#issuecomment-3145690
-    with open(os.path.join(here, rel_path)) as fp:
-        return fp.read()
-
-
-def get_version(rel_path: str) -> str:
-    for line in read(rel_path).splitlines():
-        if line.startswith("__version__"):
-            # __version__ = "0.9"
-            delim = '"' if '"' in line else "'"
-            return line.split(delim)[1]
-    raise RuntimeError("Unable to find version string.")
-
-
-long_description = read("README.rst")
+from setuptools import setup
 
 setup(
-    name="pip",
-    version=get_version("src/pip/__init__.py"),
-    description="The PyPA recommended tool for installing Python packages.",
-    long_description=long_description,
-    license="MIT",
-    classifiers=[
-        "Development Status :: 5 - Production/Stable",
-        "Intended Audience :: Developers",
-        "License :: OSI Approved :: MIT License",
-        "Topic :: Software Development :: Build Tools",
-        "Programming Language :: Python",
-        "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3 :: Only",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
-        "Programming Language :: Python :: 3.10",
-        "Programming Language :: Python :: 3.11",
-        "Programming Language :: Python :: 3.12",
-        "Programming Language :: Python :: Implementation :: CPython",
-        "Programming Language :: Python :: Implementation :: PyPy",
-    ],
-    url="https://pip.pypa.io/",
-    project_urls={
-        "Documentation": "https://pip.pypa.io",
-        "Source": "https://github.com/pypa/pip",
-        "Changelog": "https://pip.pypa.io/en/stable/news/",
-    },
-    author="The pip developers",
-    author_email="distutils-sig@python.org",
-    package_dir={"": "src"},
-    packages=find_packages(
-        where="src",
-        exclude=["contrib", "docs", "tests*", "tasks"],
-    ),
-    package_data={
-        "pip": ["py.typed"],
-        "pip._vendor": ["vendor.txt"],
-        "pip._vendor.certifi": ["*.pem"],
-        "pip._vendor.requests": ["*.pem"],
-        "pip._vendor.distlib._backport": ["sysconfig.cfg"],
-        "pip._vendor.distlib": [
-            "t32.exe",
-            "t64.exe",
-            "t64-arm.exe",
-            "w32.exe",
-            "w64.exe",
-            "w64-arm.exe",
-        ],
-    },
     entry_points={
         "console_scripts": [
             "pip=pip._internal.cli.main:main",
             "pip{}=pip._internal.cli.main:main".format(sys.version_info[0]),
             "pip{}.{}=pip._internal.cli.main:main".format(*sys.version_info[:2]),
```

Comment:
Does this `pip3.x` entry point still make sense?

In the pip wheel, it definitely does not, and actually varies from one release to the other, depending on which python version the maintainer doing the release was using.

I'd say we can make these entry points static and keep `pip` and `pip3` only.

Reply:
I agree in principle, but I don't think this PR is the right place to make this change. I specifically tried to avoid changing the metadata and focus on the format.

(Also, I received some pushback on this in https://github.com/pypa/pip/pull/11547#issuecomment-1296318295, and I don't want to add anything controversial to the PR.)

Reply:
I was not aware of that [hackery](https://github.com/pypa/pip/blob/ff05e4224bd7d1030e46d07e3a9c9419d39292fe/src/pip/_internal/operations/install/wheel.py#L275) to generate the pip console scripts. Interesting.

If I read that correctly, whatever `pip` console entry points we put in the wheels are ignored. So it's probably better to be correct and not generate the 3.x one which is going to be wrong most of the time.

> I agree in principle, but I don't think this PR is the right place to make this change. I specifically tried to avoid changing the metadata and focus on the format.

Yet this PR changes the behaviour of `setup.py`, right? if anyone was still using `setup.py install` to install pip, this PR will break it, possibly silently? So I'd say it's better to go all the way and remove `setup.py` instead of keeping a broken one.

Reply:
> Yet this PR changes the behaviour of `setup.py`, right? if anyone was still using `setup.py install` to install pip, this PR will break it, possibly silently?

I don't think so. I just tried it, seems to be working fine.

Reply:
Ok, that's right, setup.py reads the config from pyproject.toml, so that's good.

And keeping the 3.x entry point is valid for folks using, say, `build` + `installer` in the same environment. Although I still think we should aim at removing it that can be done in other PR.

Reply:
So, are we good now?